doc/manual.html

```diff
@@ -3156,6 +3162,122 @@ <h2 id="addons">Addons</h2>
     looked up through <a href="#getMode"><code>getMode</code></a>.</p>
 </section>
 
+<section id="vimapi">
+    <h2>VIM Mode API</h2>
+
+    <p>CodeMirror has a robust VIM mode that attempts to faithfully 
+    emulate VIM's most useful features. It can be enabled by 
+    including <a href="../keymap/vim.js"><code>keymap/vim.js</code>
+    </a> and setting the <code>keymap</code> option to 
+    <code>vim</code>.</p>
+
+    <h3 id="vimapi_configuration">Configuration</h3>
+
+    <p>VIM mode accepts configuration options for customizing
+    behavior at run time. These methods can be called at any time
+    and will affect all existing CodeMirror instances unless
+    specified otherwise. The methods are exposed on the
+    <code><strong>CodeMirror.Vim</strong></code> object.</p>
+
+    <dl>
+      <dt id="vimapi_setOption"><code><strong>setOption(name: string, value: any, ?cm: CodeMirror, ?cfg: object)</strong></code></dt>
+      <dd>Sets the value of a VIM option. <code>name</code> should
+      be the name of an option. If <code>cfg.scope</code> is not set
```

Comment:
There's a subtlety that for `:set`, setOption gets called twice, with and without cm, so even for local-only options you need to expect cm undefined and silently return. I'm not sure if that behavior was intentional. It leads to some ambiguity, and could probably just be fixed at the `:set` level to detect local-only options, but otherwise it should be documented.